### ISSUE TITLE
Problems in Gaussian'94 basis set parser

### ISSUE THREAD

**Issue by susilehtola** [pyscf/pyscf#635]
The Gaussian'94 parser isn't working properly. This can be exemplified with a simple input
```
#!/usr/bin/python3                                                                                                                                                                            

from pyscf import gto, scf
from pyscf.gto.basis.parse_gaussian import load as load_gaussian

mol = gto.M(
    spin = 1,
    atom = '''                                                                                                                                                                                
H 0.0 0.0 0.0                                                                                                                                                                                 
''',
    basis={
        'H': load_gaussian('hgbs.gbs', 'H')
    }
)

method = scf.UHF(mol)
method.kernel()
```
and the basis set `hgbs.gbs` which is simply one s function with exponent 1.0 and contraction coefficient 1.0
```
H 0
S 1 1.0
1.0 1.0
****
```
Trying to run this produces the indecipherable error
```
$ python3 hgbs.py 
Traceback (most recent call last):
  File "hgbs.py", line 6, in <module>
    mol = gto.M(
  File "/home/work/pyscf/pyscf/gto/mole.py", line 105, in M
    mol.build(**kwargs)
  File "/home/work/pyscf/pyscf/gto/mole.py", line 2307, in build
    self._basis = self.format_basis(_basis)
  File "/home/work/pyscf/pyscf/gto/mole.py", line 2392, in format_basis
    return format_basis(basis_tab)
  File "/home/work/pyscf/pyscf/gto/mole.py", line 425, in format_basis
    or not isinstance(atom_basis[0][0], int)):
IndexError: list index out of range
```
and it would be nice if the error message was clearer.

To make the the input work, you need to add at least **two** empty lines to the top of the basis set file, **and** add an extra `****` which normally terminates an atomic basis:
```


****
H 0
S 1 1.0
1.0 1.0
****
```

In short, there are two problems

1. error message about basis set parsing failure should be clearer
2. the Gaussian basis parser should not require extra padding and `****` to work.

**Comment by sunqm** [comment]
Bug fixed in 6a8dec012

**Comment by susilehtola** [comment]
Thanks!
